shared/params.go

```diff
@@ -380,10 +380,22 @@ func ParseProductParam(r *http.Request) (product *Product, err error) {
 // It parses the 'products' parameter, split on commas, and also checks for the (repeatable)
 // 'product' params.
 func ParseProductsParam(r *http.Request) (ProductSpecs, error) {
-	productParams := ParseRepeatedParam(r, "product", "products")
+	repeatedParam := r.URL.Query()["product"]
+	pluralParam := r.URL.Query().Get("products")
+	// Replace nested ',' in the label part with a placeholder
+	nestedCommas := regexp.MustCompile("(\\[[^\\]]*),")
```

Comment:
Use

    `pattern`

so that you don't need to escape all the backslashes.

Reply:
Done.